test/unit/mriController.test.js

```diff
@@ -0,0 +1,343 @@
+                atlasLabelSet: 'SampleLabelSet',
+                token: 'jnqpincpienfcpewnfcpewn123'
+            },
+            value: 0,
+            validationErrors: function() {
+                console.log('Testing!');
+            }
+        };
+        let validate = sinon.stub(req, 'validationErrors').resolves(true);
+        let resSpy = sinon.spy();
+
+        let res = {
+            status: sinon.stub().returns({ send: resSpy }),
+        };
+        await mriController.validator(req, res, function () {
+            console.log('Unit Test finished!');
+        });
+        assert.notDeepStrictEqual(validate.calledOnce, true);
+        assert.notDeepStrictEqual(resSpy.calledOnce, true);
+        assert.notDeepStrictEqual(resSpy.calledWith(null), true);
+        sinon.restore();
+    });
+
+    it('validatorPost function() should perform the validations correctly', async function () {
+        let reqSpy = sinon.spy();
+        let urlSpy = sinon.spy();
+        let req = {
+            body: {
+                url: 'abc.com',
+                atlasName: 'MyAtlas',
+                atlasProject: 'Visualisation@',
+                atlasLabelSet: 'SampleLabelSet',
+                token: 'jnqpincpienfcpewnfcpewn123'
+            },
+            value: 0,
+            validationErrors: function() {
+                return;
+            },
+            checkBody: sinon.stub().returns({ notEmpty: reqSpy, isURL: urlSpy })
+        };
+        let resSpy = sinon.spy();
+
+        let res = {
+            status: sinon.stub().returns({ send: sinon.stub().returns({ end: resSpy })}),
+        };
+        await mriController.validatorPost(req, res, () => {});
+        assert.strictEqual(reqSpy.callCount, 1);
+        assert.strictEqual(urlSpy.callCount, 1);
+        assert.strictEqual(resSpy.callCount, 0);
+        sinon.restore();
+    });
+
+    it('download MRI function should download the file when correct input is given', async function() {
+        let req = {
+            db: db,
+            query: {
+                url: 'https://brainbox.pasteur.fr/mri?url=https://s3.amazonaws.com/fcp-indi/data/Projects/ABIDE_Initiative/Outputs/freesurfer/5.1/CMU_a_0050642/mri/T1.mgz'
+            },
+            dirname: __dirname.split('/test')[0],
+            headers: {},
+            user: {
+                username: ''
+            },
+            session: {
+                returnTo: ''
+            },
+            originalUrl: '',
+            isTokenAuthenticated: true,
+            tokenUsername: '',
+            isAuthenticated: function() {
+                return this.isTokenAuthenticated;
+            },
+            connection: {
+                remoteAddress: 'http://localhost:3000'
+            }
+        };
+        let arr = [];
+        let authenticated = sinon.stub(req, 'isAuthenticated').resolves(true);
+        let res = {
+            render: sinon.spy()
+        };
+        await mriController.mri(req, res);
```

Comment:
`mriController.mri` is not an async function and does not return a promise, so the await will not work

Reply:
I've handled this now..

Reply:
ok the function is async now